### PR TITLE
Improve Versions in Axon Ivy Market

### PR DESCRIPTION
**Pull request by alexsuter**
- Never cut "-SNASPHOT" off
- By default never show -SNAPSHOT versions
- Add link to show -SNAPSHOT version, keep state of this link as cookie
- By default show -SNAPSHOT version of matching Axon Ivy Designer if needed

image

I can only merge this, as soon as we have released demo-projects properly.

### CODE REVIEW

**Review by ivy-lli — APPROVED** [review]
nice 👍

File: src/app/pages/market/ProductAction.php [hunk 1]
```diff
@@ -35,40 +36,68 @@ public function __invoke($request, $response, $args)
...
     $mavenArtifacts = [];
     $docUrl = '';
+    $versionsToDisplay = null;
+
+    $showDevVersionCookie = $request->getCookieParams()['showDevVersions'] ?? 'false';
+    $showDevVersions = filter_var($showDevVersionCookie, FILTER_VALIDATE_BOOLEAN) ?? false;
+    if (isset($request->getQueryParams()['showDevVersions'])) {
+      if ($request->getQueryParams()['showDevVersions'] == "true") {
+        $showDevVersions = true;
+        setcookie('showDevVersions', "true", time()+60*60*24*30 , "/");
```
Comment (ivy-cst) [inline src/app/pages/market/ProductAction.php:50]: May be even higher? Or do you renew it?

**Review by ivy-cst — APPROVED** [review]
Cool thanks ....

File: src/app/pages/market/ProductAction.php (hunk 1 above)
Comment (alexsuter) [reply src/app/pages/market/ProductAction.php:50]: uups copied. 30days. no I don't renew it. should be higer. yes.

File: src/app/pages/market/ProductAction.php [hunk 2]
```diff
@@ -100,7 +129,10 @@ public function __invoke($request, $response, $args)
       'openApiUrl' => $openApiUrl,
       'version' => $productVersion,
       'docUrl' => $docUrl,
-      'installNow' => $installNow
+      'installNow' => $installNow,
+      'versionsToDisplay' => $versionsToDisplay,
+      'switchVersion' => $showDevVersions ? "hide dev versions" : "show dev versions",
```
Comment (ivy-cst) [inline src/app/pages/market/ProductAction.php:134]: I would vote for developer instead of just dev?!?

Comment (alexsuter) [reply src/app/pages/market/ProductAction.php:134]: eigentlich steht dev für development version?!

**Pull request merged**